README.md

```diff
@@ -1 +1,303 @@
+
+
+
+## Architecture
+
+![](/images/em-arch.png)
+
+To ensure the capability of service discovery in some edge devices with low-version kernels or low-version iptables, EdgeMesh adopts the userspace mode in its implementation. In addition, it also comes with a lightweight DNS server.
+
+- Through the capability of list-watch on the edge of KubeEdge, EdgeMesh monitors the addition, deletion and modification of metadata (e.g., services and endpoints), and then creates iptables rules based on services and endpoints
+- EdgeMesh uses the same ways (e.g., Cluster IP, domain name) as the K8s Service to access services
+- When client's requests accessing a service reach a node with EdgeMesh, it will enter the kernel's iptables at first
+- The iptables rules previously configured by EdgeMesh will redirect requests, and forward them all to the port 40001 which is occupied by the EdgeMesh process (data packets from kernel mode to user mode)
+- After requests enter the EdgeMesh process, the EdgeMesh process completes the selection of backend Pods (load balancing occurs here), and then sends requests to the host where the Pod is located
+
+
+
+
+## Functionality
+
+|       Feature        |     Sub-Feature     | Edgemesh 0.1 |
+| :------------------: | :-----------------: | :----------: |
+|  Service Discovery   |                     |     `✓`      |
+|  Traffic Governance  |        HTTP         |     `✓`      |
+|                      |         TCP         |     `✓`      |
+|                      |      Websocket      |     `✓`      |
+|                      |        HTTPS        |     `✓`      |
+|     Load Balance     |       Random        |     `✓`      |
+|                      |     Round Robin     |     `✓`      |
+|                      | Session Persistence |     `✓`      |
+|   External Access    |                     |     `✓`      |
+| Multi-NIC Monitoring |                     |     `✓`      |
+
+**Noting:**
+
+- `✓` Features supported by the EdgeMesh version 
+- `+` Features not available in the EdgeMesh version, but will be supported in subsequent versions
+- `-` Features not available in the EdgeMesh version, or deprecated features
+
+
+
+## Operation Guidance
+
+#### Constraints
+In its implementation, Edgemesh has borrowed from [istio's](https://istio.io/latest/zh/docs/ops/deployment/requirements/) VirtualService, DestinationRule, and GateWay, so there are some requirements in its use:
+
+- Due to the lack of underlying CNI capabilities, when using edgemesh's capabilities, the Pod is required a hostPort (as shown in following examples)
+- while using DestinationRule, the name of the DestinationRule must be equal to the name of the corresponding Service. Edgemesh will determine the DestinationRule in the same namespace according to the name of the Service
+- Service ports must be named. The key/value pairs of port name must have the following syntax: name: \<protocol>[-\<suffix>]
```

Comment:
just describe what need to be done before starting to use EdgeMesh

Reply:
I have modified this content as required.